odoo/odoo
# [FIX] stock: prevent date_deadline from merging negative moves

**How to reproduce:**
- Unarchive MTO route
- Create product P1, storable, with MTO route selected, add vendor in vendors list.
- Create S1 for 5 units of P1.
- Create S2 for 5 units of P1.
~ S1 & S2 should be linked to the same PO
- Confirm PO
- On S1, change Delivery Date (commitment_date) in the tab Other Info
- On S2, change Delivery Date to another date.
- On S1, change the ordered qty to 4.
\> A new Transfer from Customer to Stock should have been created

https://watch.screencastify.com/v/tARlsQ7jSa8JxIZlTk3h

With the usage of the MTO route, 2 unrelated Sale Orders' transfers can be linked in the following way:
![image](https://user-images.githubusercontent.com/29302288/187923978-be5766db-6180-4074-8051-7dac2e6904fc.png)
Updating the commitment_date in S1 will:
Update date_deadline of S1's stock moves.
Propagate the update of date_deadline to the PO's stock moves.
Propagate the update of date_deadline to S2's stock moves.
\>\> S2 will hence have an incompatibility between its commitment_date (that wasn't updated) and the date_deadline field of its stock_move.

This incompatibility will prevent any merge when updating S2 ordered quantities.

OPW-2893467

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:




--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
